src/java.base/share/classes/java/util/UUID.java

```diff
@@ -181,6 +181,46 @@ public static UUID nameUUIDFromBytes(byte[] name) {
         return new UUID(md5Bytes);
     }
 
+    /**
+     * Static factory to retrieve a type 7 (time based) {@code UUID} based on
+     * the current timestamp, enhanced with sub-millisecond precision.
+     *
+     * The {@code UUID} embeds the current Unix timestamp in milliseconds into
+     * the first 6 bytes, sets the version and variant bits as per RFC 9562,
+     * replaces the first 12 random bits with a value derived from the current
+     * system clock's sub-millisecond precision, and fills the
+     * remaining bytes with cryptographically strong random data.
+     *
+     * @return A {@code UUID} generated from the current system time
+     *
+     * @spec RFC 9562
+     */
+    public static UUID timestampUUID() {
+        long msTime = System.currentTimeMillis();
+        long nsTime = System.nanoTime();
```

Comment:
I is my understanding that `System#nanoTime()` does not provide the nanosecond fraction and is not guaranteed to have nanosecond resolution. You'd have to look at `VM#getNanoTimeAdjustment(long)` to get the nanotime adjustment. However the resolution of this is OS dependent, usually microseconds, likely 10 microseconds on Windows.